### ISSUE TITLE
Bad Screenshare Quality on M1 iPads

### ISSUE THREAD

**Issue by TeaObvious** [bigbluebutton/bigbluebutton-mobile#44]
### Hardware
Apple iPad Pro 11-inch 3rd Generation
iPad OS 15.5 (19F77)
Produced: Summer 2011

### Tested Mobile App Versions
10 + 11 + 12

### Problem
The Screenshare Quality looks like 240p Streams or worse - Camera streams look normal _(The provided Screenshots are from Version 11)_
![screenshot-a7702b13](https://user-images.githubusercontent.com/8144976/174297771-a4276522-f4fc-4dc3-b641-1cda2b3e9f36.png)
![screenshot-94d22d92](https://user-images.githubusercontent.com/8144976/174297774-4c2d21f2-af26-4180-98f4-3f8779849578.png)

### Expectations
The Screenshare Quality should be similar to Airplay _(Images are Taken on the Device, not via Airplay, the Airplay quality is very similar thou)_

![2022-06-17-141755](https://user-images.githubusercontent.com/8144976/174297990-44abcdb5-a5d9-44ff-b076-afa0d96e85f9.jpg)
![2022-06-17-141633](https://user-images.githubusercontent.com/8144976/174297993-6f0ae140-e94f-4c08-8fce-b59a523f7e99.jpg)

### Steps to reproduce
1. Use an iPad Pro 11-inch with a M1 Processor
2. Install the BBB Mobile App
3. Connect to the demo-ios Server
4. Create a Session and share the Screen

Ref: https://github.com/bigbluebutton/bigbluebutton/issues/10051#issuecomment-1145473089

**Comment by TeaObvious** [comment]
Of course, as you see there are always 2 screenshots per section "Problem" + "Expectations" - so in total 4 Screenshots, lets enumerate them from 1-4 top to bottom.
In both sections the first screenshots (1 and 3) is the normal view without the popup open, the second screenshot (2 and 4) is just for reference that this also happens with popups.

**Comment by TiagoJacobs** [comment]
Ok, now I understood.

Yes, it seems we need to improve the resolution accordingly with the broadcaster ipad's capacity, it's probably going to come in a future release, keeping this open so we don't forget.

Thanks again for reporting.

**Comment by gustavo-em** [comment]
Hello, I've tested different resolutions and memory usage, please see:

100px width:
![100](https://user-images.githubusercontent.com/59415818/175374157-1d23e007-a253-4bdf-8d6d-e5930cdd9a4d.png)

200px width:
![200](https://user-images.githubusercontent.com/59415818/175374200-aaa86d74-cb2f-4201-986f-564464f0f967.png)

300px width:
![300](https://user-images.githubusercontent.com/59415818/175374259-47bfc804-06df-482e-95aa-c69acec273d1.png)

400px width:
![400](https://user-images.githubusercontent.com/59415818/175374284-d6a5430c-2cbd-4cf7-844b-05ed0a2f71ed.png)

500px width:
![500](https://user-images.githubusercontent.com/59415818/175374354-9f20d822-12ec-42d5-8774-ed8504e41651.png)

600px width:
![600](https://user-images.githubusercontent.com/59415818/175374401-8130277a-ec85-427c-9522-3657844a600f.png)

700px width:
![700](https://user-images.githubusercontent.com/59415818/175374448-fd11fc7a-3b86-4eea-8bf8-951c59f5bc12.png)

800px width:
![800](https://user-images.githubusercontent.com/59415818/175374493-ffcad2a6-8e86-4a07-b75e-db0714b740bc.png)

**Comment by TiagoJacobs** [comment]
Hello @TeaObvious - as mentioned, in our test devices, the memory limit seems to be 50MB, so it looks like the maximum we can go is 800px width.

Are you a developer with a mac + xcode?

I ask it because if this is the case, we could show you how to run the code from your mac (connected to your ipad M1), to see if this memory limit is different in ipad M1.

**Comment by TeaObvious** [comment]
@TiagoJacobs not yet, but my request for a MacBook Pro is approved so in the near future i should have access to a mac to perform this kind of task for you. I have no eta yet, you know such things can take a while in universities. :)

**Comment by TiagoJacobs** [comment]
@TeaObvious sure, please keep us posted, meanwhile, we'll try to increase it a little bit (now it's defaulting to 600px width), it seems that something around 800px will be safe.

@gustavo-em is doing more tests.

**Comment by dluschan** [comment]
Hello.
I confirm that the quality on the iPad Pro (12.9-inch 3rd generation iPadOS 16.3.1) + BigBlueButton Server 2.6.0-rc.9 (1020) is terrible. Practically nothing to make out.
Sadly ;(

**Comment by TiagoJacobs** [comment]
Hello @dluschan - we're aware of this situation.

This is caused by a memory limit on the broadcast extension on iOS.

We're (as time permits) investigating ways to improve this software, including but not limited to using a hardware encoder.

Keep watching this issue, we'll post new here as soon as we get this improved.

**Comment by agru** [comment]
Hello, any news about this issue?

**Comment by RolKl** [comment]
Hello, first of all, thank you very much for the app.
Is there any news about the problem with the low resolution?
How did the others (Zoom, Webex) solve it?

**Comment by TiagoJacobs** [comment]
Hello, we're not actively working in this issue, because we found out that to solve it it will require 2 months of work.

So we're currently bidding some customers to see if we can get the funding to implement it.

I'll post here if when get any update.

Thank all of you for the feedback.
